packages/storage_client/lib/src/fetch.dart

```diff
@@ -98,31 +108,15 @@ class Fetch {
       filename: file.path,
       contentType: contentType,
     );
-    final request = http.MultipartRequest(method, Uri.parse(url))
```

Comment:
The deleted code here is the same as in `_handleBinaryFileRequest()` so I added additional `_handleFileRequest()` instead to reduce code duplication. These two can now use an actual `_handleMultipartRequest()` method, which just accepts a `MultipartFile` which is independent of a passed `File` or `Uint8List`.